autogluon/utils/tabular/ml/models/tabular_nn/tabular_nn_model.py

```diff
@@ -96,6 +96,10 @@ def __init__(self, path: str, name: str, problem_type: str, objective_func, stop
         self._architecture_desc = None
         self.optimizer = None
         self.verbosity = None
+        if self.stopping_metric is not None and self.objective_func == roc_auc and self.stopping_metric == log_loss:
+            self.stopping_metric = roc_auc  # NN is overconfident so early stopping with logloss can halt training too quick
```

Comment:
See my comment re-updating abstract_model predict_proba, I don't think this is necessary after that change. You would just have to compute normalize_pred_proba_stopping_metric during training and pass that into predict_proba.

Reply:
This is not because of anything being broken.  This PR would've fixed the old NN issue without this change.  This change is because NN should not be early stopping based on logloss.  Logloss early stopping can make NN stop training too early, because NN tends to get overconfident very quickly (which is not a problem for AUC metric). Metrics like accuracy, AUC can still improve on validation set even after logloss has peaked.